examples/recording_service/leveldb-integration/cpp/README.md

```diff
@@ -0,0 +1,1077 @@
+cmake -DCONNEXTDDS_DIR=<connext directory> 
+      -DCONNEXTDDS_ARCH=<connext architecture>
+      -DCMAKE_BUILD_TYPE=Release 
+      -DLEVELDB_DIR=<leveldb directory>
+      -DLEVELDB_INCLUDE=<leveldb include directory>
+      ..
+cmake --build . --config Release
+```
+
+> **Note**:
+>
+> When using a multi-configuration generator, make sure you specify
+> the `--config` parameter in your call to `cmake --build .`. In general,
+> it's a good practice to always provide it.
+
+In case you are using Windows x64, you have to add the option -A in the cmake
+command as follows:
+
+```bash
+cmake -DCONNEXTDDS_DIR=<connext dir> 
+      -DCONNEXTDDS_ARCH=<connext architecture>
+      -DCMAKE_BUILD_TYPE=Release 
+      -DBUILD_SHARED_LIBS=ON .. 
+      -A x64
+```
+
+> **Note for building on Windows**:
+>
+> In file LevelDbReader.cxx, notice the first include, `dds/dds.hpp`. When
+> building on Windows, it is recommended to include that header before any 
+> direct inclusion of C, `dds_c/*.h` headers. If not, unresolved symbol link 
+> errors could happen. Also, if you need to access C headers directly, it is 
+> preferred to use header `ndds/ndds_c.h` than specific headers.
+> The reason behind this is that header `dds/core/macros.hpp` has to be called
+> before any C header is used. Otherwise, there may be a situation where symbols
+> are not correctly dll-imported.
+
+**Cross-compilation**.
+
+When you need to cross-compile the example, the above command will not work, the
+assigned compiler won't be the cross-compiler and errors may happen when linking
+against the cross-compiled Connext binaries. To fix this, you have to create a
+file with the architecture name and call CMake with a specific flag called
+``-DCMAKE_TOOLCHAIN_FILE``. An example of the file to create with the toolchain
+settings (e.g. for an ARM architecture):
+
+```cmake
+set(CMAKE_SYSTEM_NAME Linux)
+set(toolchain_path "<path to>/arm-bcm2708/gcc-linaro-arm-linux-gnueabihf-raspbian")
+set(CMAKE_C_COMPILER "${toolchain_path}/bin/arm-linux-gnueabihf-gcc")
+set(CMAKE_CXX_COMPILER "${toolchain_path}/bin/arm-linux-gnueabihf-g++")
+```
+
+Then you can call CMake like this:
+
+```bash
+cmake -DCONNEXTDDS_DIR=<connext dir> 
+      -DCMAKE_TOOLCHAIN_FILE=<toolchain file created above>
+      -DCONNEXTDDS_ARCH=<connext architecture>
+      -DCMAKE_BUILD_TYPE=Release 
+      -DLEVELDB_DIR=<leveldb directory> 
+      -DLEVELDB_INCLUDE=<leveldb include directory> 
+      ..
+```
+
+## Running *Recorder*
+
+First of all, you need to make sure that the `LevelDbWriter` shared library
+is in the system's library path.  On Mac OS X, you must use the variable
+`RTI_LD_LIBRARY_PATH` instead of the common `DYLD_LIBRARY_PATH`.
+
+This example will work with any DDS Topic discovered on the DDS Domain ID of 
+your choice. Run any application publishing data into that domain, for example
+*RTI Shapes Demo* or *RTI DDS Ping*.
+
+The supplied XML configuration for *Recorder* expects you to set an environment 
```

Comment:
I would not use the enviornment word here, since the variable can be set with multiple ways. I recommend setting using it the -D command-line option

Reply:
Yes, that's true. I do prefer the -D command-line option too.